monitor/ConsoleReporter.js

```diff
@@ -72,16 +72,20 @@ define(function(require) {
 				// IE8 has console.log and JSON, so we can make a
 				// reasonably useful warn() from those.
 				// Credit to webpro (https://github.com/webpro) for this idea
-				if (typeof localConsole.log ==='function'
-					&& typeof JSON !== 'undefined') {
-					log = warn = function (x) {
-						if(typeof x !== 'string') {
+				// typeof localConsole.log will return 'object' in IE8, so can't test it with === 'function'
+				// Since this is more of a corner case for IE8, I'm ok to check it with !== 'undefined' to reduce complexity
+				if (typeof localConsole.log !== 'undefined' && typeof JSON !== 'undefined') {
```

Comment:
Does it fail because console is not available in IE8? The code already checked for that so if that is the cause, the code is fine. 

Homa
Increase the Net Worth of your time

> On Mar 17, 2015, at 9:42 AM, John Babak <notifications@github.com> wrote:
> 
> In monitor/ConsoleReporter.js:
> 
> > @@ -72,16 +72,20 @@ define(function(require) {
> >  				// IE8 has console.log and JSON, so we can make a
> >  				// reasonably useful warn() from those.
> >  				// Credit to webpro (https://github.com/webpro) for this idea
> > -				if (typeof localConsole.log ==='function'
> > -					&& typeof JSON !== 'undefined') {
> > -					log = warn = function (x) {
> > -						if(typeof x !== 'string') {
> > +				// typeof localConsole.log will return 'object' in IE8, so can't test it with === 'function'
> > +				// Since this is more of a corner case for IE8, I'm ok to check it with !== 'undefined' to reduce complexity
> > +				if (typeof localConsole.log !== 'undefined' && typeof JSON !== 'undefined') {
> Just accessing a property on the console object (localConsole.log here and localConsole.error, localConsole.dir above) may throw an exception in IE8, and the whole library will terminate its execution.
> 
> \
> Reply to this email directly or view it on GitHub.
>